extensions/objects/templates/unicode-string-editor.component.spec.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for unicode string editor.
+ */
+
+import { EventEmitter, SimpleChanges } from '@angular/core';
+import { ExternalSaveService } from 'services/external-save.service';
+import { FormsModule } from '@angular/forms';
+import { UnicodeStringEditorComponent } from './unicode-string-editor.component';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+describe('UnicodeStringEditorComponent', () => {
+  let component: UnicodeStringEditorComponent;
+  let fixture: ComponentFixture<UnicodeStringEditorComponent>;
+  let externalSaveService: ExternalSaveService;
+
+  var externalSaveEventEmitter = new EventEmitter();
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [UnicodeStringEditorComponent]
+    }).compileComponents();
+  }));
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UnicodeStringEditorComponent);
+    component = fixture.componentInstance;
+    externalSaveService = TestBed.get(ExternalSaveService);
+    spyOnProperty(externalSaveService, 'onExternalSave').and.returnValue(
+      externalSaveEventEmitter);
+    component.alwaysEditable = false;
+    component.value = 'random value';
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should initialize component when interaction rule is selected', () => {
+    spyOn(component, 'closeEditor').and.callThrough();
+    spyOn(component.componentSubscriptions, 'add');
+    component.active = true;
+
+    expect(component.active).toBe(true);
+
+    component.ngOnInit();
+
+    expect(component.componentSubscriptions.add).toHaveBeenCalled();
+    expect(externalSaveService.onExternalSave).toEqual(
+      externalSaveEventEmitter);
```

Comment:
Are we expecting to have some changes here? (Also, I think we should focus on testing the behavior instead of making sure that the function is getting called, so if we can test the behavior changes by this function call, we don't need to ensure that whether the function got called or not and we don't need to write a code-comment stating that we cannot assert the function call.)

Reply:
I have tested the behaviour as a separate test.

https://github.com/oppia/oppia/blob/93c03aaabc98bb6ff7e4594f87a68e6135e60d0f/extensions/objects/templates/unicode-string-editor.component.spec.ts#L91-L109